folding-schemes/src/folding/nova/circuits.rs

```diff
@@ -100,23 +117,18 @@ where
     #[allow(clippy::type_complexity)]
     pub fn hash(
         self,
-        crh_params: &CRHParametersVar<CF1<C>>,
+        sponge: &PoseidonSpongeVar<CF1<C>>,
```

Comment:
Could this be agnostic to the specific hash? (if it's too complicated/messy don't worry about it, we can keep it as it is for the moment)